### PR TITLE
[connectors] Upsert and backfill Zendesk brands and categories as `data_source_folders`

### LINKED ISSUE #9161: [KW-Search] Update & backfill `folders` for zendesk
See [#8527](https://github.com/dust-tt/dust/issues/8527)

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Closes [#9161](https://github.com/dust-tt/dust/issues/9161);
- Add API calls whenever a Zendesk brand is upserted/deleted in connectors (3 folders per brand bc 1 brand, 1 HC, 1 tickets).
- Same for the categories.
- Add a backfill script to backfill existing Zendesk brands and categories.
- Tested locally, I get the rows in `data_sources_folders`, I also get the nodes in `data_sources_nodes` with `mime_type` `application/vnd.dust.folder`.

## Risk

- Relatively low, not a lot of user-facing impact as of now and easily rollbackable.

## Deploy Plan

- Deploy connectors.
- Run migration

### CODE REVIEW

File: connectors/src/connectors/zendesk/temporal/gc_activities.ts [hunk 1]
```diff
@@ -244,20 +251,19 @@ export async function removeEmptyCategoriesActivity(connectorId: number) {
     dataSourceId: dataSourceConfig.dataSourceId,
   };
 
-  const categoryIds = (
-    await ZendeskCategoryResource.fetchIdsForConnector(connectorId)
-  ).map(({ categoryId }) => categoryId);
+  const categoryAndBrandIds =
```
Comment (philipperolet) [inline connectors/src/connectors/zendesk/temporal/gc_activities.ts:254]: nit: this naming makes beleive that you can get ids either from categories or brands
hard to name but maybe `cateroryIdsWithBrand`

**Review by philipperolet — APPROVED** [review]
LGTM :+1: Left a nit

File: connectors/src/connectors/zendesk/temporal/gc_activities.ts (hunk 1 above)
Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/temporal/gc_activities.ts:254]: indeed I like that better 👍

Comment (aubin-tchoi) [reply connectors/src/connectors/zendesk/temporal/gc_activities.ts]: done

**Pull request merged**